app/lib/api/file_system/file_system_dav.dart

```diff
@@ -202,18 +202,36 @@ class DavRemoteDocumentFileSystem extends DocumentRemoteSystem {
     if (!forceSync && remote.hasDocumentCached(path)) {
       cacheContent(path, data);
     }
-    var last = path.lastIndexOf('/');
-    if (last == -1) {
-      last = path.length;
-    }
+
+    final directoryPath = p.dirname(path);
     // Create directory if not exists
-    final directoryPath = path.substring(0, last);
-    if (!await hasAsset(directoryPath)) {
-      await createDirectory(directoryPath);
+    final directory = Directory(directoryPath);
+    if (!await directory.exists()) {
+      await directory.create(recursive: true);
     }
+
     final response =
-        await createRequest(path.split('/'), method: 'PUT', bodyBytes: data);
-    if (response?.statusCode != 201 && response?.statusCode != 204) {
+        await createRequest(p.split(path), method: 'PUT', bodyBytes: data);
+    if (response?.statusCode == 201 || response?.statusCode == 204) {
+      // File updated
+      return;
+    } else if (response?.statusCode == 409) {
+      // Conflict, rename the file
+      final baseName = p.basenameWithoutExtension(path);
+      final extension = p.extension(path);
+      String newPath = p.join(directoryPath, '$baseName(1)$extension');
```

Comment:
This is a "updateFile" function. It should override the path if it already exists

Reply:
no you need to rename, because when you create a file using quick launch and don't give it a name it leads to errors

Reply:
That is been handled inside the create file method: https://github.com/LinwoodDev/Butterfly/blob/a79aa7777f6a3a9a0768567cf7292520d3c40bb9/app/lib/api/file_system/file_system.dart#L180

It tries to find the next unique name (`My file.bfly`, `My file (1).bfly`, `My file (2).bfly`) and runs `updateFile` after that.

But `updateFile` will also be run if you open an existing document and click on the save button. Here we want to override the current file.

So in conclusion we want `updateFile` to always override the current file if it exists.

Reply:
ok then I avoid managing 409 to rename the file. it actually works

Reply:
i did it like this ` @override
  Future<void> updateFile(String path, List<int> data,
      {bool forceSync = false}) async {
    // Creazione della directory se non esiste
    final directoryPath = p.dirname(path);
    final directory = Directory(directoryPath);
    if (!await directory.exists()) {
      await directory.create(recursive: true);
    }

    // Request to overwrite the file
    final response =
        await createRequest(p.split(path), method: 'PUT', bodyBytes: data);
    if (response?.statusCode == 200 ||
        response?.statusCode == 201 ||
        response?.statusCode == 204) {
      // File overwritten successfully
      return;
    } else {
      // Management of error cases
      throw Exception(
          'Failed to update document: ${response?.statusCode} ${response?.reasonPhrase}');
    }
  }` works correctly, except in the case where the user deliberately creates a file with the same name as another file, in which case you get `Unhandled Exception: Exception: Failed to update document: 500 Internal Server Error` perhaps the best way would be to not handle it and warn the user with a popup that they need to rename the file

Reply:
Yeah maybe, but which webdav server do you use? On nextcloud it works like a charm

Reply:
i use nextcloud